src/Symfony/Component/Yaml/Parser.php

```diff
@@ -101,6 +115,16 @@ public function parse($value, $flags = 0)
             mb_internal_encoding('UTF-8');
         }
 
+        Inline::$tagResolver = $this->tagResolver;
+        try {
+            return $this->doParse($value, $flags);
+        } finally {
+            Inline::$tagResolver = null;
```

Comment:
woudln't it be better to pass it as argument instead of forcing to maintain some state externally ?